x-pack/legacy/plugins/siem/public/lib/telemetry/middleware.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Action, Dispatch, MiddlewareAPI } from 'redux';
+
+import { track, METRIC_TYPE, TELEMETRY_EVENT } from './';
+import { timelineActions } from '../../store/actions';
+
+export const telemetryMiddleware = (api: MiddlewareAPI) => (next: Dispatch) => (action: Action) => {
+  if (timelineActions.endTimelineSaving.match(action)) {
```

Comment:
I found that using a middleware to track based on actions was much simpler than trying to insert more `track()` calls at various points in the codebase. If we're okay with this approach, we should maybe convert the remaining `track()` calls to dispatch actions, too.

Reply:
Definitely agree -- I think this approach is the right direction for reporting client-side telemetry (when in redux-land).